### PR TITLE
Add default proposal parameters and custom proposal configuration

### PR DESCRIPTION
**Pull request by tensojka**
This changes quorum and proposal voting time to come from a storage variable and thus be updatable. Also adds support for custom proposals that have a shorter voting time.

### CODE REVIEW

**Comment by tensojka** [comment]
Still need to fix tests, but otherwise ready for review

File: src/proposals.cairo [hunk 1]
```diff
@@ -554,5 +579,15 @@ mod proposals {
             self.custom_proposal_type.write(idx, config);
             idx
         }
+
+        fn set_default_proposal_params(ref self: ComponentState<TContractState>, quorum: u32, proposal_voting_seconds: u32) {
```
Comment (scobi7) [inline src/proposals.cairo]: Looks good for proposal parameter configurations. Only potential suggestion is maybe emitting an event for changes in proposal parameters?

**Review by scobi7 — COMMENTED** [review]
Logic looks good. Im unfamiliar with the small quorum values that are hard coded so just ensuring those are good and can be left like that should be okay. (like in get quorum or the asserts in set_default_proposal_params)

Also maybe emitting an event for when parameters of proposals are changed may be a good idea.

File: src/proposals.cairo (hunk 1 above)
Comment (tensojka) [reply src/proposals.cairo]: No strong opinion on that, but don't see a reason to consume that event, so not adding

**Pull request merged**